Source/SpatialGDK/Private/SpatialPackageMapClient.cpp

```diff
@@ -170,9 +201,36 @@ FNetworkGUID FSpatialNetGUIDCache::AssignNewStablyNamedObjectNetGUID(const UObje
 void FSpatialNetGUIDCache::RemoveEntityNetGUID(worker::EntityId EntityId)
 {
 	FNetworkGUID EntityNetGUID = GetNetGUIDFromEntityId(EntityId);
-	FHashableUnrealObjectRef* ActorRef = NetGUIDToUnrealObjectRef.Find(EntityNetGUID);
-	NetGUIDToUnrealObjectRef.Remove(EntityNetGUID);
-	UnrealObjectRefToNetGUID.Remove(*ActorRef);
+	AActor* Actor = Cast<AActor>(GetObjectFromNetGUID(EntityNetGUID, false));
+	RemoveNetGUID(EntityNetGUID);
+
+	// Remove references to the actor's subobjects 
+	// Slightly hacky as this relies on offsets starting from 1 and being consecutive
```

Comment:
Could introduce future problems, but getting the subobjects here is problematic, since the UnrealMetadata component is removed from view by now.

Reply:
After discussion offline, this will be moved to OnRemoveComponent for the UnrealMetadata.